### PR TITLE
#39-datepicker-and-textarea-AdminPage

### PR DESCRIPTION
**Pull request by AndrewPang321**
- Implement DatePicker and TextArea using Angular Material in Admin Page
- Add 2 buttons for saving data
- Save relevant information into Firebase
- Chang to 40-space indentation
- Fixed indentation and redundant lines

### CODE REVIEW

**Comment by AndrewPang321** [comment]
For clarification:
- In admin.html: added 2 datePickers(start & end date), 1 textArea(details) and 2 buttons(one for date and one for textArea).
- In admin.js: 2 functions(saveDate() & saveContent()) corresponding to 2 buttons in admin.html. Both save the information to the firebase under "events/eventName/admin/param" (for details, please take a look at 36_redesign_event_1 on FIrebase. Some tested data are displayed there for testing.)

File: app/js/admin.js [hunk 1]
```diff
@@ -48,17 +56,25 @@ angular.module("teamform-admin-app", ["firebase", "ngMaterial"])
...
         $scope.param.$save();
         // Finally, go back to the front-end
         window.location.href = "index.html";
-    }
+    };
+    $scope.saveDate = function() {
+        //console.log($scope.startDate);
+        ref.update({'startDate': $scope.startDate.toString(), 'endDate': $scope.endDate.toString()});
+    };
+    $scope.saveContent = function() {
```
Comment (chautsunman) [inline app/js/admin.js:94]: combine $scope.saveContent and $scope.saveDate function (1 save button for saving both the details and the dates)

File: app/js/admin.js [hunk 2]
```diff
@@ -7,8 +7,16 @@ $(document).ready(function() {
     }
 });
 
-angular.module("teamform-admin-app", ["firebase", "ngMaterial"])
+angular.module("teamform-admin-app", ["firebase", "ngMaterial", "ngMessages"])
 .controller("AdminCtrl", function($scope, $firebaseObject, $firebaseArray) {
+
+    // Date
+    $scope.startDate = new Date();
```
Comment (chautsunman) [inline app/js/admin.js]: display the saved dates if the event was created

File: app/js/admin.js [hunk 3]
```diff
@@ -7,8 +7,16 @@ $(document).ready(function() {
...
+    $scope.endDate = new Date();
+
+    // TextArea: details
+    $scope.details;
```
Comment (chautsunman) [inline app/js/admin.js]: display the saved details if the event was created

File: app/admin.html [hunk 4]
```diff
@@ -44,17 +44,42 @@ <h2 id="text_event_name"></h2>
...
+                    </div>
+                    <br>
+                    <button type="button" class="btn btn-primary" ng-click="saveDate()" >Save Date</button>
+                    <br><br>
+                    <!-- TextArea -->
+                    <md-input-container class="md-block">
+                        <label>Details: </label>
+                        <textarea ng-model="details" md-maxlength="150" rows="2" md-select-on-focus></textarea>
+                    </md-input-container>
+                    <button type="button" class="btn btn-primary" ng-click="saveContent()" >Save Details</button>
```
Comment (chautsunman) [inline app/admin.html]: change `type="button"` into `type="submit"`

File: app/admin.html [hunk 5]
```diff
@@ -44,17 +44,42 @@ <h2 id="text_event_name"></h2>
...
                     <button><span class="glyphicon glyphicon-minus" ng-click="changeMaxTeamSize(-1);"></span></button>
                     <br><br>
                     <button type="button" class="btn btn-primary" ng-click="saveFunc();" >Create/Modify</button>
+                    <!-- Datepicker -->
+                    <br><br>
+                    <div class="row" ng-cloak>
+                        <div layout-gt-xs="row">
+                            <div flex-gt-xs>
+                                <h4>Start Date: </h4>
+                                <md-datepicker ng-model="startDate" md-placeholder="Enter date"></md-datepicker>
```
Comment (chautsunman) [inline app/admin.html]: change this field to be required using `ng-required="true"`

File: app/admin.html [hunk 6]
```diff
@@ -44,17 +44,42 @@ <h2 id="text_event_name"></h2>
...
+                            </div>
+
+                            <div flex-gt-xs>
+                                <h4>End Date: </h4>
+                                <md-datepicker ng-model="endDate" md-placeholder="Enter date"></md-datepicker>
```
Comment (chautsunman) [inline app/admin.html]: change this field to be required using `ng-required="true"`

File: app/admin.html [hunk 7]
```diff
@@ -44,17 +44,42 @@ <h2 id="text_event_name"></h2>
...
+                            </div>
+                        </div>
+                    </div>
+                    <br>
+                    <button type="button" class="btn btn-primary" ng-click="saveDate()" >Save Date</button>
+                    <br><br>
+                    <!-- TextArea -->
+                    <md-input-container class="md-block">
+                        <label>Details: </label>
+                        <textarea ng-model="details" md-maxlength="150" rows="2" md-select-on-focus></textarea>
```
Comment (chautsunman) [inline app/admin.html]: change this field to be required using `ng-required="true"`

**Comment by chautsunman** [comment]
display the saved details and dates in the event team page (eventteam.html)

**Comment by AndrewPang321** [comment]
Done.
- Modify admin.js: combine two functions into one & use
  $scope.startDate instead of $scope.startDate.toString() for displaying
  the correct date format on eventteam.html
- modify admin.html(change start/endDate and Details to ng-require="true") & eventteam.js(load info from firebase)
- modify eventteam.html: display event’s information under the
  EventName & using the correct date format with filtering to display startDate &
  endDate

**Pull request merged**